hackathon_site/hardware/models.py

```diff
@@ -15,8 +15,29 @@ class Meta:
     def __str__(self):
         return self.name
 
+class AnnotatedHardwareManager(models.Manager):
+    def get_queryset(self):
+        return (
+            super()
+                .get_queryset() 
+                .annotate(
+                    quantity_checked_out=Count(
+                        "order_items",
+                        filter=(
+                            Q(order_items__part_returned_health__isnull=True)
+                            & ~Q(order_items__order__status="Cancelled")
+                        ),
+                        distinct=True,
+                    )
+                )
+                .annotate(
+                    quantity_remaining=(F("quantity_available") - F("quantity_checked_out"))
+                )
+        )
 
 class Hardware(models.Model):
```

Comment:
Ok after digging into the `refresh_from_db()` stuff, let's update the model to be this:

```python
class Hardware(models.Model):
    objects = AnnotatedHardwareManager()

    class Meta:
        verbose_name_plural = "hardware"

    class Config:
        annotated_fields = ("quantity_checked_out", "quantity_remaining")

    ...

    created_at = models.DateTimeField(auto_now_add=True, null=False)
    updated_at = models.DateTimeField(auto_now=True, null=False)

    def refresh_from_db(self, using=None, fields=None):
        super().refresh_from_db(using, fields)

        # Fetch the queryset again to populate the annotations. This is somewhat
        # inefficient since it performs the query twice, so use sparingly.
        db_instance_qs = self.__class__.objects.filter(pk=self.pk)
        db_instance = db_instance_qs.get()

        for field_name in self.Config.annotated_fields:
            setattr(self, field_name, getattr(db_instance, field_name))

    def __str__(self):
        return self.name
```

I added a `Config` class, and overwrote `refresh_from_db`. Since no matter what we'd have to do a separate query anyway to get the updatde fields, might as well make this method use it.